lib/reply.js

```diff
@@ -120,26 +120,28 @@ function wrapOnSendEnd (err, payload) {
 }
 
 function onSendEnd (reply, payload) {
+  var headersSent = reply.res.headersSent
```

Comment:
I think we can bail out from this function here. If headers has been sent, then whatever we do here is worthless. I think onSend hooks can be run twice in error conditions, eg a hook errors. However we pass through here even if the res emits errors. I think we should just check this in the pump callback and avoid all of this.